### DISCUSSION TITLE
Connecting to Mongo using certificate files sslCA option

### DISCUSSION THREAD

**Discussion by egonzal93** [prisma/prisma#9898] [Q&A]
Here's an example connection using native Mongo. Is there a way to implement this on Prisma?

Not mentioned here:

https://www.prisma.io/docs/getting-started/setup-prisma/add-to-existing-project/mongodb/connect-your-database-typescript-mongodb/
```js
var MongoClient = require('mongodb').MongoClient,
   fs = require('fs');

var ca = [fs.readFileSync(__dirname + "cacert.pem")];

MongoClient.connect("mongodb://dumper:dumper@example.1.dblayer.com:10900,example.0.dblayer.com:10901/dbname", {
         mongos: {
            ssl: true,
            sslValidate: true,
            sslCA: ca,
            poolSize: 1,
            reconnectTries: 1
         },
      }, function(err, db) {
         ...
```

**Comment by ryands17** [comment]
@egonzal93 👋 

It should be possible to pass it in the connection string options as shown [here](https://docs.mongodb.com/manual/reference/connection-string/#connection-options).

**Reply by egonzal93** [reply]
I never seen sslCA passed as an array to a mongo db connection string. 

I have a GraphQL server using mongoose and I am doing it this way and its working beautifully.

const mongoUri  = 'mongodb://example_mongoUser:example_password123@servername.foo1.net:37017,ervername.foo2.net:37017/test?ssl=true&authSource=admin&replicaSet=MONGO_SET1234';

const devCert = fs.readFileSync(path.resolve(__dirname, '../config/certs/DEV.pem'),);
const uatRootCert = fs.readFileSync(path.resolve(__dirname, '../config/certs/UAT.pem'),);
const prodRootCert = fs.readFileSync(path.resolve(__dirname, '../config/certs/PROD.pem'),);

const mongoOptions = {
  useUnifiedTopology: true,
  useCreateIndex: true,
  useNewUrlParser: true,
  sslValidate: true,
  sslCA: [prodRootCert, uatRootCert, devCert]
}

const conn = await mongoose.connect(mongoUri, mongoOptions);

I wanted to use Prisma but it does NOT tell me how to implement a connection like above. 

Is there an example code on Prisma to do this?

**Reply by ryands17** [reply]
It's shown [here](https://docs.mongodb.com/manual/reference/connection-string/#mongodb-urioption-urioption.tlsCAFile) in the docs.

**Comment by Ahmadre** [comment]
I also have the same problem and cannot connect with prisma including mongo options.

I need to create this client:

https://www.mongodb.com/docs/manual/core/csfle/fundamentals/automatic-encryption/#enabling-automatic-client-side-field-level-encryption

NOT POSSIBLE!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
